### ISSUE TITLE
Can not find -lpthread for android ndk arm64-v8a build

### ISSUE THREAD

**Issue by Nick-infinity** [google/glog#893]
Hello, 
I am trying to build bazel/example:main with ndk r25 tool chain for android api level 31 . I am using bazel version 6.0.0 . When I try to compile the the target  I get linker error for pthread library. In my understanding ndk does not have a separate pthread library. 
I am suspecting : https://github.com/google/glog/blob/f545ff5e7d7f3df95f6e86c8cb987d9d9d4bd481/bazel/glog.bzl#L73
and
https://github.com/google/glog/blob/f545ff5e7d7f3df95f6e86c8cb987d9d9d4bd481/src/config.h.cmake.in#L34

**Comment by drigz** [comment]
> Are there any specific instructions to build for android using ndk for bazel ?

No, I'm not sure if anybody has tried that and can't see any issues referencing it.

If you can work out how to break the dependency on pthread, please send a PR to change glog.bzl to use appropriate cflags on Android: https://github.com/google/glog/blob/f545ff5e7d7f3df95f6e86c8cb987d9d9d4bd481/bazel/glog.bzl#L145

**Comment by jiabei-dm** [comment]
This is because glog depends on "@com_github_gflags_gflags//:gflags" which links pthread. Instead, Android targets should depend on "@com_github_gflags_gflags//:gflags_nothreads".
